cli/tests/pcluster/cli/test_commands.py

```diff
@@ -333,3 +343,117 @@ def test_setup_bucket_with_resources_upload_failure(
         cluster._upload_instance_types_data()
 
     check_bucket_mock.assert_called_with()
+
+
+@pytest.mark.parametrize(
+    (
+        "head_object_side_effect",
+        "get_object_side_effect",
+        "get_object_return_value",
+        "expected_result",
+        "expected_exception",
+    ),
+    [
+        (
+            pytest.param(
+                None,
+                None,
+                {
+                    "Body": Mock(
+                        read=Mock(
+                            return_value=json.dumps({"bootstrapped_features": ["basic", "export-logs"]}).encode("utf-8")
+                        )
+                    )
+                },
+                True,
+                None,
+                id="The bootstrap file exists, content is in the new format, and contains all required features",
+            )
+        ),
+        (
+            pytest.param(
+                None,
+                None,
+                {
+                    "Body": Mock(
+                        read=Mock(return_value=json.dumps({"bootstrapped_features": ["basic"]}).encode("utf-8"))
+                    )
+                },
+                False,
+                None,
+                id="The bootstrap file exists, content is in the new format, but some required features are missing",
+            )
+        ),
+        (
+            pytest.param(
+                None,
+                None,
+                {"Body": Mock(read=Mock(return_value="bucket is configured successfully.".encode("utf-8")))},
+                False,
+                None,
+                id="The bootstrap file exists, content is in old format (not JSON string)",
+            )
+        ),
+        (
+            pytest.param(
+                AWSClientError(function_name="head_object", message="Not Found", error_code="404"),
+                None,
+                None,
+                False,
+                None,
+                id="The bootstrap file does not exist (head_object throws 404 error)",
+            )
+        ),
+        (
+            pytest.param(
+                None,
+                AWSClientError(function_name="get_object", message="Access Denied", error_code="403"),
+                None,
+                None,
+                AWSClientError,
+                id="get_object throws AWSClientError (not a 404 error)",
+            )
+        ),
+        (
+            pytest.param(
+                None,
+                None,
+                {"Body": Mock(read=Mock(return_value="{invalid json}".encode("utf-8")))},
+                False,
+                None,
+                id="The bootstrap file content cannot be parsed as JSON",
+            )
+        ),
+    ],
+)
+def test_check_bucket_is_bootstrapped(
```

Comment:
This unit test should be moved to `cli/tests/pcluster/models/test_s3_bucket.py` because the original function 
`check_bucket_is_bootstrapped()` is in file `cli/src/models/s3_bucket`.

Reply:
Done, thanks for catching this.